bin/echo.c

```diff
@@ -45,9 +45,11 @@ int negotiate(struct s2n_connection *conn)
     s2n_blocked_status blocked;
     do {
         if (s2n_negotiate(conn, &blocked) < 0) {
-            fprintf(stderr, "Failed to negotiate: '%s'. %s\n", s2n_strerror(s2n_errno, "EN"), s2n_strerror_debug(s2n_errno, "EN"));
-            fprintf(stderr, "Alert: %d\n", s2n_connection_get_alert(conn));
-            S2N_ERROR_PRESERVE_ERRNO();
+            if (s2n_error_get_type(s2n_errno) != S2N_ERR_T_BLOCKED) {
```

Comment:
Won't this excessively call `s2n_negotiate` when it is blocked? Would it be better to add a `poll` instead of calling it immediately?

Reply:
I've updated this code to prevent a loop.